contracts/RewardsPool.sol

```diff
@@ -0,0 +1,253 @@
+        __Ownable_init();
+
+        alphaNumerator = 1;
+        alphaDenominator = 3;
+        settings = _settings;
+    }
+
+    /**
+     * @dev update the settings.
+     */
+    function setSettings(ISettings _settings) external onlyOwner {
+        settings = _settings;
+    }
+
+    /**
+     * @dev Update the alpha for cobb-douglas function.
+     * @param _alphaNumerator int32.
+     * @param _alphaDenominator int32.
+     */
+    function setAlpha(int32 _alphaNumerator, int32 _alphaDenominator) public onlyOwner {
+        require(_alphaNumerator > 0 && _alphaDenominator > 0, "!alpha");
+        alphaNumerator = _alphaNumerator;
+        alphaDenominator = _alphaDenominator;
+
+        emit Alpha(alphaNumerator, alphaDenominator);
+    }
+
+    /**
+     * @dev get the Pool reward by deploymentId, era and indexer. returns my labor and total reward.
+     * @param deploymentId byte32.
+     * @param era uint256.
+     * @param indexer address.
+     */
+    function getReward(bytes32 deploymentId, uint256 era, address indexer) public view returns (uint256, uint256) {
+        Pool storage pool = pools[era].pools[deploymentId];
+        return (pool.labor[indexer], pool.totalReward);
+    }
+
+    /**
+     * @dev Add Labor(reward) for current era pool.
+     * @param deploymentId byte32.
+     * @param indexer address.
+     * @param amount uint256. the labor of services.
+     */
+    function labor(bytes32 deploymentId, address indexer, uint256 amount) external {
+        require(amount > 0, 'Invalid amount');
+        IERC20(settings.getSQToken()).safeTransferFrom(msg.sender, address(this), amount);
+
+        uint256 era = IEraManager(settings.getEraManager()).safeUpdateAndGetEra();
+        EraPool storage eraPool = pools[era];
+        Pool storage pool = eraPool.pools[deploymentId];
+        // deployment created firstly.
+        if (pool.totalReward == 0) {
+            eraPool.unclaimDeployment += 1;
+        }
+        // indexer joined the pool firstly.
+        if (pool.labor[indexer] == 0) {
+            eraPool.indexerUnclaimDeployments[indexer] += 1;
+
+            IStaking staking = IStaking(settings.getStaking());
+            uint256 myStake = staking.getTotalStakingAmount(indexer);
+            require(myStake > 0, 'No indexer');
+            pool.stake[indexer] = myStake;
+            pool.totalStake += myStake;
+        }
+        pool.labor[indexer] += amount;
+        pool.totalReward += amount;
+        pool.unclaimTotalLabor += amount;
+        pool.unclaimReward += amount;
+
+        emit Labor(deploymentId, indexer, amount, pool.totalReward);
+    }
+
+    /**
+     * @dev Collect reward (stake) from previous era Pool.
+     * @param deploymentId byte32.
+     * @param indexer address.
+     */
+    function collect(bytes32 deploymentId, address indexer) external {
+        uint256 currentEra = IEraManager(settings.getEraManager()).safeUpdateAndGetEra();
+        require(currentEra > 0, 'Waiting Era');
```

Comment:
Looks don't need to check this as era starts from `1`

Reply:
Yeah, Removed.